pkg/kubectl/cmd/set/set_image.go

```diff
@@ -220,7 +225,11 @@ func (o *ImageOptions) Run() error {
 		}
 
 		info.Refresh(obj, true)
-		cmdutil.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, false, "image updated")
+
+		if len(o.Output) > 0 {
+			return o.PrintObject(o.Cmd, o.Mapper, obj, o.Out)
+		}
+		cmdutil.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, o.DryRun, "image updated")
```

Comment:
At some point, we need a way to get a printerfunc in complete that doesn't require tons of gorp.